include/alpaka/rand/Philox/helpers/metaprog.hpp

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2021 Jiri Vyskocil
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <functional>
+#include <numeric>
+#include <type_traits>
+
+namespace alpaka
+{
+    namespace rand
+    {
+        namespace helpers
+        {
+            // --- static_for --------------------------------------------
+
+            /** Recursive helper for implementing unrolled for loops.
+             *
+             *  @tparam First Low iteration bound
+             *  @tparam Last High iteration bound
+             */
+            template<int First, int Last>
+            struct static_for
+            {
+                template<typename Fn>
+                constexpr void operator()(Fn&& fn) const
+                {
+                    static_assert(First < Last, "Invalid static loop bounds");
+                    fn(First);
+                    static_for<First + 1, Last>()(fn);
+                }
+            };
+
+            /// End of recursion for \a static_for
+            template<int N>
+            struct static_for<N, N>
+            {
+                template<typename Fn>
+                constexpr void operator()(Fn&& fn) const
+                {
+                }
+            };
+
+            /** @brief \a static_for for functions without parameters
+             *
+             *  Calls the function repeatedly, but doesn't pass the counter into the loop.
+             */
+            template<unsigned N>
+            struct repeat_n
+            {
+                template<typename Fn>
+                constexpr void operator()(Fn&& fn) const
+                {
+                    static_for<0, static_cast<int>(N)>()([&](int i) { fn(); });
+                }
+            };
```

Comment:
This is basically a weaker `boost::mp11::for_each`. The entire file should be moved into `alpaka/meta`. Furthermore, I want to again raise the question whether alpaka should adopt a proper metaprogramming library like boost::mp11, so we do not need to write and maintain such utilities ourselves.

Reply:
Yes. I didn't know where to keep these, I'll move it to alpaka/meta. I'd vote for mp11 (or something similar) though.